File: 392-is-subsequence/392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if t == s or s == "":
            return True
        if(len(s) > len(t)):
            return False
        j = 0
        for i in range(len(t)):
            if j == len(s):
                break
            if s[j] == t[i]:
                j+=1
        if j == len(s):
            return True
        return False

File: 392-is-subsequence/392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int i = 0, j = 0;
        while(i < s.length() && j < t.length()){
            if(s.charAt(i) != t.charAt(j)){
                j++;
            } else{
                i++;
                j++;
            }
        }
        if(i == s.length())
            return true;
        return false;
    }
}